Added the top-dmg command

supinko opened the pull request:
top-dmg shows you the top ten tanks by average damage, for example:

Blitzbot: @usagi8 [-INV-], 
1: 2076.08 dmg - T57 Heavy Tank (usa, 10)
2: 1836.13 dmg - T54E1 (usa, 9)
3: 1796.01 dmg - Leopard 1 (germany, 10)
4: 1782.45 dmg - IS-8 (ussr, 9)
5: 1780.01 dmg - ST-I (ussr, 9)
6: 1779.82 dmg - AMX 50 120 (france, 9)
7: 1765.41 dmg - AMX 50 100 (france, 8)
8: 1729.45 dmg - M48 Patton (usa, 10)
9: 1697.32 dmg - Bat.-Châtillon 25 t AP (france, 9)
10: 1680.47 dmg - T30 (usa, 9)

supinko committed:
Added average damage to tank-win-rate and win-rate

supinko committed:
Added the top-dmg command

CodeMan99 commented:
What do you think about trying to add a little history to this command as well? I'm thinking make use of emojis to indicate movement on the list.

```
@usagi8 [-INV-],
1 :arrow_up: 2076.08 - T57 Heavy Tank (usa, 10)
2 :arrow_right: 1940.22 - T54E1 (usa, 9)
3 :arrow_double_down: 1700.42 - Leopard PTA (germany, 9)
```

The history here is position only, not value change. So the storage can be nice and simple.

```
{sentMsg: sent, topDamageOrder: [13569, 18997, 16897, ..., 9745]}
```

Where the array holds `tank_id` values.

supinko committed:
Addressed code review comments.

Added a number formatting function to make the non-percentage numbers in the win-rate command look better.

supinko committed:
Added a timestamp to the log messages so it's easier to debug problems.

supinko committed:
Merge branch 'add-avg-dmg' into add-top-dmg

supinko commented:
> What do you think about trying to add a little history to this command as well? I'm thinking make use of emojis to indicate movement on the list.
> 
> ```
> @usagi8 [-INV-],
> 1 :arrow_up: 2076.08 - T57 Heavy Tank (usa, 10)
> 2 :arrow_right: 1940.22 - T54E1 (usa, 9)
> 3 :arrow_double_down: 1700.42 - Leopard PTA (germany, 9)
> ```
> The history here is position only, not value change. So the storage can be nice and simple.
> 
> ```
> {sentMsg: sent, topDamageOrder: [13569, 18997, 16897, ..., 9745]}
> ```
> Where the array holds `tank_id` values.

That could be an interesting enhancement that encourages more usage of the command. Let me note that as a TODO.

supinko committed:
Moved the formatNumber function out to helpers.js and applied some new formatting to the maxXp command to make it more readable. If you disagree with the new formatting, I can revert it.

supinko committed:
Merge branch 'add-avg-dmg' into add-top-dmg

supinko committed:
Added some nicer formatting to the top-dmg command output.

supinko committed:
Removed user name from log message. Moved log and error functions out of helpers.js and into bin/bot.js.

supinko committed:
Merge branch 'add-avg-dmg' into add-top-dmg
